torch/_dynamo/variables/torch.py

```diff
@@ -828,7 +828,7 @@ def get_comparable_state(state):
                 )
             )
 
-        def speculate_subgraph(f, sub_args, graph_checkpoint, checkpoint):
+        def old_speculate_subgraph(f, sub_args, graph_checkpoint, checkpoint):
```

Comment:
Oh no what happened that these split?

Reply:
I cannot actually rewrite Dynamo capture for map() and cond() to use the new `speculate_subgraph`. This is because we do not support capturing nn.Modules or nn.Module attributes yet. That situation is tricky because there is an open question of if the nn.Module should appear in the body function or if we should completely trace out the nn.Module so that it does not.

I'm leaving that as a follow-up, but let me know if you want it in this PR.

Reply:
Followup is fine, but let's add a comment :)